### ISSUE TITLE
type D should not allow length

### ISSUE THREAD

**Issue by arcanist123** [abaplint/transpiler#411]
such a statement should produce syntax error. It is instead working
``` abap
DATA tsl TYPE d length 21 decimals 7.
```

**Comment by arcanist123** [comment]
I reopened it in abaplint and we can close it here since the transpiler is generating the date type correctly
